File: test/Proxy.Tests/ProxyDiscovererTests.Compile.cs
using System;

namespace Moq.ProxyDiscovererTests
{
    public class Tests
    {
        public void WhenMockingFormatterThenCanInvokeIt()
        {
            var mock1 = Mock.Of<ICustomFormatter>();
            var mock2 = Mock.Of<ICustomFormatter>();

            var result1 = mock1.Format("Hello {0}", "World", null);
            var result2 = mock2.Format("Hello {0}", "World", null);
        }
    }
}

File: test/Proxy.Tests/ProxyDiscovererTests.Compile.vb
Imports System
Imports Moq

Public Class Tests
    Public Sub WhenMockingFormatterThenCanInvokeIt()
        Dim mock1 As ICustomFormatter = Mock.Of(Of ICustomFormatter)
        Dim mock2 As ICustomFormatter = Mock.Of(Of ICustomFormatter)

        Dim result1 As String = mock1.Format("Hello {0}", "World", Nothing)
        Dim result2 As String = mock2.Format("Hello {0}", "World", Nothing)
    End Sub
End Class
